applications/alarm/alarm-plugins/org.csstudio.alarm.beast/src/org/csstudio/alarm/beast/client/AlarmTreePV.java

```diff
@@ -39,7 +58,76 @@ public class AlarmTreePV extends AlarmTreeLeaf
     private volatile String current_message = SeverityLevel.OK.getDisplayName();
 
     private volatile String value = null;
+	
+	/**
+	 * Thread-safe PV reader for the limit PVs. For each interesting PV limit an
+	 * instance is created and a thread initiated for the listener.
+	 */
+	private class PV {
+		private final ExecutorService PVINFO_THREADPOOL = Executors.newCachedThreadPool();
```

Comment:
```suggestion
	private static class PV {
		private static final ExecutorService PVINFO_THREADPOOL = Executors.newCachedThreadPool();
```

Doesn't this create a new threadpool for every single new leaf-node in the alarm tree? If so that seems massively inefficient compared to using a single static threadpool (and will probably also leak threads as the alarms change, as I don't see that we're calling `shutdown()` on these threadpools anywhere)?

For comparison, pretty much _all_ of the channel access in the GUI goes through a single threadpool at the `PVManager` level...

You can see this quite clearly in e.g. VisualVM, I was able to get the GUI to over 2000 threads (!) just by switching to my local instrument a few times...

![image](https://github.com/user-attachments/assets/b30daea1-d79f-41f2-a858-c7d50b3f8c5c)

`cachedThreadPool` does kill off unused threads eventually, but we still leak the underlying executor object.

(In addition, please could we give the threads descriptive names so that we can tell where they were created from when looking in stack traces/visualvm/debugger later)

Reply:
Done